### PR TITLE
Unpublish a conversation when the name is changed

### PR DESCRIPTION
**Pull request by nicolocaruso85**
Related to #194

### CODE REVIEW

File: src/ConversationBuilder/Observers/ConversationObserver.php [hunk 1]
```diff
@@ -20,14 +22,29 @@ public function created(Conversation $conversation)
...
-    public function updated(Conversation $conversation)
+    public function updating(Conversation $conversation)
     {
-        //
+        // Unpublish the conversation from DGraph.
+        if ($conversation->isDirty('name')) {
+            if ($conversation->getOriginal('status') == 'published') {
+                $originalName = $conversation->getOriginal('name');
+
+                $dGraph = new DGraphClient(env('DGRAPH_URL'), env('DGRAPH_PORT'));
```
Comment (stu-greenshoots) [inline src/ConversationBuilder/Observers/ConversationObserver.php]: Don't instantiate DGraphClient like this - it should be pulled from the service layer

**Comment by stu-greenshoots** [comment]
Would be even better for the DGraph dependency to be injected into the Conversation builder observer if possible, or if not, loaded globally in a constructor mether

**Comment by nicolocaruso85** [comment]
fixes https://github.com/opendialogai/core/issues/114

**Comment by stu-greenshoots** [comment]
Closing this as it should be picked up as part of https://github.com/opendialogai/core/issues/194

**Pull request closed without merging**